### PR TITLE
Use section instead of header in dashboard pages

### LINKED ISSUE #14840: Dashboard pages shouldn't have two banner landmarks
**Describe the bug**

On the user dashboard there are two "banner" landmarks:

- The `<header>` at the top of the page (with logo, search, etc)
- The `<header>` containing "Total post reactions", "Total post views" etc

The same issue is observable across the other dashboards also (e.g dashboard/following_tags).

On each dashboard, [we should only have one banner landmark on the page](https://dequeuniversity.com/rules/axe/4.3/landmark-no-duplicate-banner?application=AxeChrome). While the wrapper for the "total post reactions" etc can just be changed to a `div`, this content should also be moved to live inside the `<main>` element, and this change might need a little bit of CSS tweaking to make sure everything continues to look consistent.

**To Reproduce**

- Log in to the app
- Go to `/dashboard`
- Open up the dev tools and run the axe extension
- Notice the error "Document should not have more than one banner landmark"

**Expected behavior**

- The error should not appear in the axe report
- Visually everything should remain the same
- The same should apply to the other dashboard pages

**Screenshots**

Screenshot of the dashboard with the axe error highlighted and the header area highlighted

**Additional context**

Each dashboard page could be tackled in separate PRs if this keeps things more manageable.

### PR DESCRIPTION
**Pull request by MohamedAsan**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Accessibility

## Description
Change `<header>` to `<section>` to avoid having two banner landmarks and move it under <main>

## Related Tickets & Documents
Resolves #14840

## QA Instructions, Screenshots, Recordings
Verified that there are no visual changes in both desktop(Chrome/Safari/Firefox) & mobile browsers(Chrome/Safari)
Axe Results

### UI accessibility concerns?
Verified by using screen reader

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: 
- [ ] I need help with writing tests

### CODE REVIEW

**Comment by MohamedAsan** [comment]
retest this please

**Comment by MohamedAsan** [comment]
@aitchiss @cmgorton Kindly review this

**Comment by cmgorton** [comment]
@MohamedAsan someone will likely review your PR after the weekend. We do not typically work on the weekends 😊Thanks for your contribution so far.

**Comment by msarit** [comment]
@MohamedAsan Hello! 👋🏾 
Could you please merge `main` into your branch? It will resolve the TravisCI failure 👍🏾

**Comment by MohamedAsan** [comment]
Thank you @benhalpern

**Comment by rhymes** [comment]
Hi @MohamedAsan, there's now a conflict, can you take a look?

File: app/views/dashboards/followers.html.erb [hunk 1]
```diff
@@ -3,10 +3,10 @@
 <div id="index-container"
   data-params="<%= params.to_json(only: %i[which per_page]) %>"
   data-which="followers">
-    <header class="crayons-layout">
-      <h1 class="p-2 pb-0 m:p-0 crayons-title">Dashboard &raquo; Followers</h1>
+    <section class="crayons-layout" aria-labelledby="header-title">
+      <h1 class="p-2 pb-0 m:p-0 crayons-title" id="header-title">Dashboard &raquo; Followers</h1>
       <%= render "actions_mobile" %>
-    </header>
+    </section>
```
Comment (aitchiss) [inline app/views/dashboards/followers.html.erb:9]: I think we need to make a few further tweaks to get the landmarks all in the place we want -

- This section is part of the main content of the page, so let's move it inside the `main`
- Excellent plan labelling the section, but I don't think this is necessary in all cases - when a section is labelled, it's added to the landmark regions for screen reader users. In this case I don't think the section is sufficiently important/standalone to merit that, so we can just remove the `aria-labelledby`

The same can be applied to the other areas below too

**Review by aitchiss — CHANGES_REQUESTED** [review]
Thanks for taking this one on! It's looking good, but left a note about some further tweaks I think we need to make - let me know if you have any questions at all!

File: app/views/dashboards/followers.html.erb (hunk 1 above)
Comment (MohamedAsan) [reply app/views/dashboards/followers.html.erb]: Thanks for the detailed explanation. Learnt a few things here :) Honestly, I don't know much about accessibility, and I was just resolving the errors shown by axe. I have made the changes as requested. Kindly let me know if those are okay

Comment (aitchiss) [reply app/views/dashboards/followers.html.erb]: Thanks for making those changes! It's great to see those errors disappearing, and this makes things much better for landmark navigation for screen reader users 😄 

If you want to check out some examples of how screen readers allow for landmark navigation, there's [a nice article summarising this on w3](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/at.html) - with your changes, people will now be able to get to the main content more easily, and not be confused by the duplicate header! 🌟

**Review by aitchiss — APPROVED** [review]
Looking great!

**Comment by cmgorton** [comment]
Thanks so much for your work here @MohamedAsan !

**Pull request merged**